include/maliput_sparse/geometry/line_string.h

```diff
@@ -199,8 +235,10 @@ class LineString final {
 
  private:
   std::vector<CoordinateT> coordinates_{};
+  std::vector<Point> points_{};
   Segments segments_{};
   double length_{};
+  std::shared_ptr<KDTreeData> kdtree_;
```

Comment:
Here we have a shared_ptr but we are calling make_unique. 

Some questions:
- Why shared_ptr?
- Why KDTreeData? I would just call it KDTree, the class namespacing will apply.

Reply:
I went for using shared_ptr because:
 - It allows us to use the default copy constructor and avoid having multiple kdtree's in memory. (if you copy the references for the kdtree, will go the same space in memory)
 - In [LaneGeometry's constructor](https://github.com/maliput/maliput_sparse/blob/fece2d445927b432c4daa54629a785f37fb85e8c/src/geometry/lane_geometry.cc#L55) we are copying the linestring. If we use unique_ptr we should move it.
 
 We could refactor this to use unique_ptr, probably the lane geometry class now will need to receive unique_ptr<LineString> and the class won't have the possibility to be copied.

Reply:
> Why KDTreeData? I would just call it KDTree, the class namespacing will apply.
Indeed

Reply:
I am not fond of shared_ptr and I think it will play against us any potential parallelization we would like to use. 
Also, I would prefer to pass raw pointers and consider a clone method so as to control copies as well. We don't want to copy points everywhere but because of that we also don't want to extend the life of the points where we don't want it to go.
Let's make a case for shared_ptr in which we don't really know where the kdtree will be used and how, and I'll be more than happy to add it.

Reply:
### Unique ptr
If going for a unique_ptr here I wouldn't add a copy constructor that regenerates the kdtree because that would be way non-performance for the class.

So I purpose switching to a unique ptr but let me tackle this in a follow-up PR because it affects so many classes here and in `maliput_osm`. LineString3d is spread across this package and at the parser level in `maliput_osm`:
 - LaneGeometry will be the class that will hold the ownership of the linestrings
 - As LineString has a non-sharable member, all LineString instances will be a `std::unique_ptr` instance, and all classes holding a LineString will also need to be a unique_ptr.
 (Note that heap-allocated memory isn't as fast as stack memory)
 
 ### Shared ptr
 
> Let's make a case for shared_ptr in which we don't really know where the kdtree will be used and how, and I'll be more than happy to add it.

`maliput_sparse` isn't a final and contained backend, but a tool where other backends, relying on other formats, will build their castle above. 
With this in mind, it is a bit hard to think ways in which LineString class will be used. HOWEVER, the case that quickly comes to my mind is the case where you want to use the same LineString to be used by two adjacent lanes.
```cpp
maliput_sparse::geometry::LineString3d left_lane_0 = FromMyFormatToMaliputSparse(weird_format_left_line_string);
maliput_sparse::geometry::LineString3d right_lane_1 = left_lane_0; // wuala, kdtree isn't built again.


@agalbachicar  ptal

Reply:
Sorry for not coming back here before.

> So I purpose switching to a unique ptr but let me tackle this in a follow-up PR because it affects so many classes here and in maliput_osm

I am OK with deferring to a follow up PR the conversion to a unique_ptr.

> With this in mind, it is a bit hard to think ways in which LineString class will be used. HOWEVER, the case that quickly comes to my mind is the case where you want to use the same LineString to be used by two adjacent lanes.

Then, it makes sense to have a view-type of the LineString that allows you to access it but not necessarily own the memory. It could be arranged somewhere else. For that to happen, we can change the builder to construct LineStrings at the Segment level, and then organize how we want them to be used. That means, we could derive LaneGeometries from the set of LineStrings or explicitly organize the pair of LineStrings at the moment of construction the LaneGeometry. This does not require a shared_ptr.

Reply:
https://github.com/maliput/maliput_sparse/issues/54